### ISSUE TITLE
Error mockup for frontmatter parsing issues

### ISSUE THREAD

**Issue by dillonkearns** [dillonkearns/elm-pages#18]
## Goal
Let's brainstorm, as a community, on how we can make a specific error message as helpful as possible! We can figure out this process for refining error messages "on paper" before we do so in code, to think about it from the user's perspective and then think about how the message will serve them.

See the notes and philosophy in the [`elm-pages` Error Message Style Guide](https://github.com/dillonkearns/elm-pages/blob/master/docs/error-style-guide.md).

## How to pitch in
You can simply write a comment giving your thoughts and suggestions, or you can try writing out an error message yourself, taking the error below as a starting point!

## Situation
This mockup is for the error the user is presented with when there is a Json decode error in parsing their frontmatter.

## Error

**CONTEXT**
```
I was trying to parse the file `content/blog/my-post.txt`

But I didn't find a `Pages.DocumentHandler` for `.txt` files in your `Pages.Platform.application` config.

You have document parsers for the following extensions:
`.md`, `.emu`, `.json`
```

**SUGGESTION**

```
So you could either:
1) Create your new file with one of those extensions, like `content/blog/my-post.md`
2) Move the file `content/blog/my-post.txt` outside of the `content` folder, or delete it
3) Add a `Pages.DocumentHandler` for `.txt` files, something like

    Pages.Document.parser
        { extension = "txt"
        , metadata = Decode.succeed ()
        , body = \_ -> Ok ()
        }
```

**EDUCATIONAL MATERIAL**
```
You can learn more about how elm-pages deals with parsing different document types,
and some techniques and best practices for managing your content files and metadata types
at elm-pages.com/docs/document-handlers
```

**Comment by dillonkearns** [comment]
Maybe the top banner should contain the filename 🤔 

```
-- UNKNOWN CONTENT EXTENSION ------------------------------------------------ content/index.md
```

**Comment by bgins** [comment]
I have a suggestion for **SUGGESTION**. I think it would be better if
```
2) Move the file `content/index.md` outside of the `content` folder, or delete it
```
was the last suggestion.

It is definitely a way to stop seeing the error, but it doesn't give a clear path for adding content, which is probably what the user intends. Putting it last would give it a bit less importance and make it seem like more of a "last resort".

**Comment by tennety** [comment]
Since the user has already created the file, maybe reword `1. Create your new file with one of those extensions` to `1. Change the extension of 'content/index.md' from '.md' to one of these extensions`?

**Comment by tennety** [comment]
Suggestion for `EDUCATIONAL MATERIAL`:

**Comment by dillonkearns** [comment]
This is really fantastic, thank you both for your great suggestions here! Very nice getting more eyes on this, I'm liking this process. Thanks for taking the time! 🙏

I'll let you know when I take a first pass at getting this error message implemented 👍
